packages/pn-personafisica-webapp/src/component/Deleghe/__test__/ConfirmationModal.test.tsx

```diff
@@ -45,21 +35,101 @@ describe('ConfirmationModal Component', () => {
+        handleClose={mockCancelFunction}
+        onConfirm={mockConfirmFunction}
+        onConfirmLabel={'Conferma'}
+        onCloseLabel={'Cancella'}
+      />
+    );
     const confirm = result!.queryAllByTestId('dialogAction')[1];
-    expect(confirm).toBeInTheDocument();
+    const cancel = result!.queryAllByTestId('dialogAction')[0];
     expect(confirm).toHaveTextContent(/Conferma/i);
+    expect(cancel).toHaveTextContent(/Cancella/i);
 
     fireEvent.click(confirm);
     expect(mockConfirmFunction).toBeCalledTimes(1);
+
+    fireEvent.click(cancel);
+    expect(mockCancelFunction).toBeCalledTimes(1);
   });
 
-  it('checks that the cancel function is executed', () => {
+  it('renders the dialog with default labels', () => {
+    useIsMobileSpy.mockReturnValue(false);
+
+    const result = render(
+      <ConfirmationModal
+        open={true}
+        title={'Test Title'}
+        handleClose={mockCancelFunction}
+        onConfirm={mockConfirmFunction}
+      />
+    );
+    const confirm = result!.queryAllByTestId('dialogAction')[1];
     const cancel = result!.queryAllByTestId('dialogAction')[0];
-    expect(cancel).toBeInTheDocument();
+    expect(confirm).toHaveTextContent(/Riprova/i);
     expect(cancel).toHaveTextContent(/Annulla/i);
+  });
 
-    fireEvent.click(cancel);
-    expect(mockCancelFunction).toBeCalledTimes(1);
+  it('renders the dialog with no confirm button', () => {
+    useIsMobileSpy.mockReturnValue(false);
+
+    const result = render(
+      <ConfirmationModal
+        open={true}
+        title={'Test Title'}
+        handleClose={mockCancelFunction}
+        onConfirm={mockConfirmFunction}
+      />
+    );
+    const cancel = result!.queryAllByTestId('dialogAction')[0];
+    expect(cancel).toHaveTextContent(/Annulla/i);
+  });
+
+  it('renders the dialog closed', () => {
+    useIsMobileSpy.mockReturnValue(false);
+
+    const result = render(
+      <ConfirmationModal
+        open={false}
+        title={'Test Title'}
+        handleClose={mockCancelFunction}
+        onConfirm={mockConfirmFunction}
+        onConfirmLabel={'Conferma'}
+        onCloseLabel={'Annulla'}
+      />
+    );
+    const dialog = result!.queryByRole('dialog');
+
+    expect(dialog).toBeNull();
+  });
+
+  it('renders the mobile view of the dialog', () => {
+    useIsMobileSpy.mockReturnValue(true);
+
+    const result = render(
+      <ConfirmationModal
+        open={true}
+        title={'Test Title'}
+        handleClose={mockCancelFunction}
+        onConfirm={mockConfirmFunction}
+        onConfirmLabel={'Conferma'}
+        onCloseLabel={'Annulla'}
+        minHeight={'6em'}
+        width={'20em'}
+      />
+    );
+    const buttons = result!.queryAllByTestId('dialogAction');
+    const dialog = result!.queryByRole('dialog');
+    const stack = result.queryByTestId('dialogStack');
+
+    expect(dialog).toBeInTheDocument();
+    expect(buttons).toHaveLength(2);
+    expect(stack).toHaveStyle('flex-direction: column');
```

Comment:
Considerare la possibilità di creare una funzione che ritorni il componente ConfirmationModal e di passare come parametri quelle proprietà che cambiano di test in test.
Poi nel render di ogni componente richiamare la funzione con i parametri corretti

Reply:
Aggiunta funzione di rendering per non ripetere i parametri che rimangono invariati